Userland/Libraries/LibIMAP/Parser.cpp

```diff
@@ -270,7 +270,7 @@ ErrorOr<void> Parser::parse_untagged()
 ErrorOr<StringView> Parser::parse_quoted_string()
 {
     dbgln_if(IMAP_PARSER_DEBUG, "p: {}, parse_quoted_string()", m_position);
-    auto str = consume_while([](u8 x) { return x != '"'; });
+    auto str = consume_while([this](u8 x) { return x == '"' ? m_buffer[m_position - 1] == '\\' : true; });
```

Comment:
Reading `m_position` in a lambda passed to `consume_while` looks pretty iffy to me. Maybe it's nicer to just do a loop here instead of using `consume_while`?

(And, same comments as in the other PR: Consider adding a test, consider adding spec comments.)

Reply:
> Maybe it's nicer to just do a loop here instead of using `consume_while`?

done

> Consider adding a test, consider adding spec comments.

I do plan on making a separate PR adding spec comments and tests for LibIMAP, I just wanted to get a few of these PRs merged first to make things a bit less crashy while testing.